### ISSUE TITLE
Can `get5_check_auths 0` be removed as an option?

### ISSUE THREAD

**Issue by nickdnk** [splewis/get5#871]
Hello

We've run into some development issues related to situations where `get5_check_auths` is `0` (disabled).

In that context, we need the community's help to figure out if we can remove this option, which means teams will always be locked based on SteamID, as we don't really see a good practical purpose for it existing, and it introduces a lot of extra "if-else" logic to the current code.

Please vote with an emoji on this issue to help us:

## Please only place one emoji and only thumbs up or down!

**Comment by danieladugyan** [comment]
It's been a while since I did the initial setup of my server, but I remember having to set `get5_check_auths 0` in order to not get kicked with the error message "You are not a player in this match". 

**TL;DR** is that I have `get5_check_auths 0` in my `get5.cfg`. I changed it for a reason when I configured my server a few years back. Maybe it's unnecessary, I don't know.

Below is a longer explanation of how I use get5. Maybe you can tell from that if the command is necessary.

I have a single self-hosted server that I use for a multitude of things: grenade practice, team scrims and in-house matches to name a few.

I use get5 because of features such as:
- Being able to lock players to correct teams (for scrims)
- Hassle free demo recording
- Round backup system

Crucially, when running scrims I will know my teammates SteamIDs in advance. My `scrim_template.cfg` is therefore configured to lock those SteamIDs to `team1` and our opponents will be placed in the other team.

However, I will **not** know any SteamIDs in advance when I'm hosting in-house matches. I can not run get5_scrim because that is configured to lock the SteamID of my team to team1 (as stated above). So instead, I use get5_creatematch when every player has joined a team.

**Comment by nickdnk** [comment]
Hello @danieladugyan 

None of the examples you mention require the use of `get5_check_auths 0`. `get5_creatematch` creates a match configuration with current players and will use their Steam IDs to lock the teams. You would only need to set `get5_kick_when_no_match_loaded 0` to allow people to join the server before loading a match.

Also, please update your plugin to version 0.10.1.

**Comment by danieladugyan** [comment]
Makes sense to me. 
I did think of another scenario. I sometimes have to rebalance teams mid-game when using CS:GO in a classroom setting. Would that still be possible if this command was removed? It's probably not be an issue that affects enough people (or is even within the scope of your project) that it's worth taking into consideration, but since you're asking for feedback I thought I'd mention it.

For me personally, I'm fine either way since I'll probably just stay on my version of the get5_plugin until it breaks.

**Comment by nickdnk** [comment]
> 

In a regular match loaded via a [match configuration](https://splewis.github.io/get5/latest/match_schema/), you can use [`get5_removeplayer`](https://splewis.github.io/get5/latest/commands/#get5_removeplayer) + [`get5_addplayer`](https://splewis.github.io/get5/latest/commands/#get5_addplayer) to move a player to another team. They will be kicked in this process though, but if done during a pause it should be fine.

If you want to do in-game swapping of players, you should be running in [scrim mode](https://splewis.github.io/get5/latest/getting_started/#scrims), where the [`get5_ringer`](https://splewis.github.io/get5/latest/commands/#get5_ringer) command will be able to easily move a player to another team.

Staying on an older version should not be your solution though, as get5 still has quite a few problems we're working to fix.

**Comment by nickdnk** [comment]
@Dying-Ducks @latisak

Please explain your use-case for the parameter.

**Comment by nickdnk** [comment]
We're leaving it in, as people seem to be using it for various things, the most important one being the fact that you cannot play CS at all using Get5 if Steam authorization fails due to problems on Steam's end. For emergency situations, you have to be able to deactivate this check, even though a lot of functionality will break at the same time.
